scantool/scantool_850.c

```diff
@@ -693,6 +701,34 @@ static void interpret_value(enum l7_namespace ns, uint16_t addr, UNUSED(int len)
 	} else if (ns==NS_LIVEDATA && addr==0x1000) {
 		/* ECU pin A4, MCU P7.4 input, divider ratio 8250/9460 */
 		printf("MAF sensor signal: %.2f V\n", (float)buf[0]*9460/8250*5/256);
+	} else if (ns==NS_LIVEDATA && addr==0x1800) {
+		printf("Short term fuel trim: %+.1f%%\n", (float)buf[0]*100/128-100);
+	} else if (ns==NS_LIVEDATA && addr==0x1900) {
+		/* possibly in units of 0.004 milliseconds (injection time) */
+		printf("Long term fuel trim, additive (unscaled): %+d\n", (signed int)buf[0]-128);
+	} else if (ns==NS_LIVEDATA && addr==0x1A00) {
+		printf("Long term fuel trim, multiplicative: %+.1f%%\n", (float)buf[0]*100/128-100);
+	/* ECU 6E: */
+	} else if (ns==NS_LIVEDATA && addr==0x0500) {
+#if 0
+		/* Bench tested, but not verified on vehicle */
+		printf("Mode selector: MS1 %s, MS2 %s, switch position %s\n", (buf[0]&1)?"low":"high", (buf[0]&2)?"low":"high", mode_selector_positions[(buf[0]>3)?4:buf[0]]);
```

Comment:
range check in the `mode_selector_positions[]` index is a bit confusing (although correct)  - I see " if >3, then 4" ; brain says "maybe he meant 'if > 3, then 3'. Maybe simplify this printf by splitting out the mode_selector_position assignment. E.g. something like `const char *mode_string = mode_sel...[...]; "
Not sure if using `sizeof` would make it clearer, too

Reply:
I've changed it to use a preprocessor macro.